parser/parser.y

```diff
@@ -11904,6 +11904,10 @@ StringType:
 	{
 		tp := $1.(*types.FieldType)
 		tp.SetCharset($2.(*ast.OptBinary).Charset)
+		if $2.(*ast.OptBinary).Charset == charset.CharsetBin {
+			tp.AddFlag(mysql.BinaryFlag)
+			tp.SetCollate(charset.CollationBin)
+		}
```

Comment:
Should this be blob? I am not sure.

Reply:
It's because of some tests. The stmt `create t(a mediumblob)` will set the col `a` to CollationBin and add binary flag. The another option is to remove two lines for `blobtype`.
https://github.com/pingcap/tidb/blob/486ee8ecd0a016fee38a0accb8b9d58ff8e15d7c/parser/parser.y#L11899-L11900

Which one do you think is better?